docs/source/cuda/cuda_array_interface.rst

```diff
@@ -69,31 +69,307 @@ The following are optional entries:
+
+  - If the Producer has no enqueued work, or work only enqueued on the stream
+    identified by ``stream``, then this condition is met.
+  - If the Producer has enqueued work on the data on multiple streams, then it
+    must enqueue events on those streams that follow the enqueued work, and
+    then wait on those events in the provided ``stream``. For example:
+
+    1. Work is enqueued by the Producer on streams ``7``, ``9``, and ``15``.
+    2. Events are then enqueued on each of streams ``7``, ``9``, and ``15``.
+    3. Waits on the events from Step 2 are enqueued into stream ``3``, and the
+       ``stream`` entry is set to ``3``.
+
+* If there is no work enqueued on the data, then the ``stream`` entry may be
+  either ``None``, or not provided.
+
+Optionally, to facilitate the User to relax the conformance to synchronization
+semantics:
+
+* Producers may provide a configuration option to always set ``stream`` to
+  ``None``.
+* Consumers may provide a configuration option to ignore the value of ``stream``
+  and act as if it were ``None`` or not provided.  This elides synchronization
+  on the Producer-provided streams, and allows enqueuing work on streams other
+  than that provided by the Producer.
+
+These options should be not be set by default in either a Producer or a
+Consumer. The exact mechanism by which these options are set, and related
+options that Producers or Consumers might provide to allow the user further
+control over synchronization behavior are not prescribed by the CAI
+specification.
+
+
+Synchronization in Numba
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Numba is neither strictly a Producer nor a Consumer - it may be used to
+implement either by a User. In order to facilitate the correct implementation of
+synchronization semantics, Numba exhibits the following behaviors related to
+synchronization of the interface:
+
+- When Numba acts as a Consumer (for example when an array-like object is passed
+  to a kernel launch): If ``stream`` is an integer, then Numba will immediately
+  synchronize on the provided ``stream``, and set the *default stream* of the
+  array to the provided stream.
+- When Numba acts as a Producer (when the ``__cuda_array_interface__`` property
+  of a Numba Device Array is accessed): If the exported Device Array has a
+  *default stream*, then it is given as the ``stream`` entry. Otherwise,
+  ``stream`` is set to ``None``.
+
+.. note:: In Numba's terminology, the *default stream* for a Device Array is a
+          property of the Device Array specifying the stream in which Numba will
+          enqueue asynchronous transfers on if no other stream is provided as an
+          argument to the function invoking the transfer. It is not the same as
+          the `Default Stream
+          <https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#default-stream>`_
+          in normal CUDA terminology.
+
+Resulting from these properties, these consequences are intended:
```

Comment:
Just a bit clearer ("these properties" is a bit unclear -- which properties? The text earlier refers to Numba's synchronization behavior, which I believe is what is meant here?)

```suggestion
Numba's synchronization behavior results in the following intended consequences:
```